client/client.go

```diff
@@ -136,6 +146,9 @@ func NewClient(cfg *config.Config) (*Client, error) {
 
 	// Start the client!
 	go c.run()
+
+	// Start the consul client
+	go c.consulClient.SyncWithConsul()
```

Comment:
Do we need synchronization around this? `go` statements without a channel or mutex are a code smell. For example it's possible for this function to get to `return c, nil` before the `SyncWithConsul` call is made. Does that matter?